onnxruntime/core/graph/graph_utils.h

```diff
@@ -86,20 +81,63 @@ bool GetRepeatedNodeAttributeValues(const Node& node,
   return false;
 }
 
-/** Removes the given Node from the Graph and keeps Graph consistent by rebuilding needed connections.
-    We support the removal of the Node as long as the following conditions hold:
-    - There should be no implicit inputs.
-    - Only one of the outputs is used by downstream operators (but it can have multiple output edges).
-    - If the Node has a single incoming node (and possibly multiple initializers), we can remove the Node and
-      connect its incoming node to its outgoing nodes.
-    - If the Node has a single initializer as input, we remove the Node and feed the initializer as input to its
-      output nodes. */
-bool RemoveNode(Graph& graph, Node& node);
+/** Tests if we can remove a node and merge its input edge (if any) with its output edges.
+Conditions:
+ - Only one of the node's outputs is used by downstream operators
+   - multiple edges for the single used output are allowed
+ - The node must have zero or one input edges 
+   - we merge any input edge with the output edges for the single output that is used
```

Comment:
I think we should note here that there might be multiple inputs if there is a single input edge (we use that one and discard the ones coming from initializers or graph inputs).